Review

MargulisMax opened the pull request.

MargulisMax committed:
Exercise 1 is done.

MargulisMax committed:
Exercise 1 - Removed comments and prints

MargulisMax committed:
Exercise 2 - Done.

MargulisMax committed:
Exercise 2 - Added relevant URLs

MargulisMax committed:
Delete exercise1.py

MargulisMax committed:
Delete exercise2.py

MargulisMax committed:
Recommiting for Review

li0nbelenky commented on home-assignments/session1/exercise1.py:
```diff
@@ -0,0 +1,61 @@
+        print(ErrVal)
+
+def SetAgeToBucketRange(age, bucranges):
+    for bucrange in bucranges:
```
There should not be hard-coded bucket values,
You should have a generic code that constructs them regardless of input.
For example, you might get buckets: [90, 100, 110, 120]

li0nbelenky commented on home-assignments/session1/exercise1.py:
```diff
@@ -0,0 +1,61 @@
+            return 'bucket_25-40'
+
+def CreateBucketRanges(bucketlist):
+    buckets = bucketlist
```
functions in python should be named 'create_bucket_ranges', not capitalised and also please use more descriptive names.

schefs reviewed (commented):
in Validate_JSON func you may want to not only print your error but to 'raise' an exception with a custom string that describes it.
in your function SetAgeToBucketRange your first if statement is ok but very long and hard to understand,  you may want to divide it into variables just for the readability
on FillTheBuckets func you are writing to a yaml file and not closing it, you shoud close it or use 'with' statement for auto closing.
code looks great very readable

schefs commented:
on exercise2 you may want to set your API key in a global variable, or set it to be injected on function call 
you could split your main to multiple functions, for example one that prints your human readable weather, and one that prints weather from a list of cities.

MargulisMax committed:
Fixed Collaborator's Remarks:
1. changed function names(upper to lower case + separating with underscore)
2. closed the file when exporting to yaml
3. regarding raise error in validate_json function, because its a TypeError exception we are not obligated to catch and re-raise them(according to python fundamentals - 09.Do not guard against type errors.)

MargulisMax committed:
Minor fix of set_age_to_bucket_range function.

MargulisMax committed:
Fixed The Following:
1. changed function's names to lower case.
2. set constant parameters instead of the long concatenate URL.

MargulisMax committed:
Finished exercise1, pre-Instructor check.

MargulisMax committed:
Fixed:
1. Parameter help messages.
2. Added space after get_weather function.
